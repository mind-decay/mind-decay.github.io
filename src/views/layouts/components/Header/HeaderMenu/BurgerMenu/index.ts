export * from './BurgerMenu';
export * from './BurgerMenu.styled';
