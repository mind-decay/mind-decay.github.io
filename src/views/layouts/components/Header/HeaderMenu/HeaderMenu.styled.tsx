import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

import { typography } from 'styles/typography';
import { screenBreakpoints } from 'styles/utils';

export const HeaderMenuWrapper = styled.div`
  display: flex;
  align-items: center;
  column-gap: 7rem;

  @media screen and (max-width: ${screenBreakpoints.tablet}px) {
    column-gap: 3rem;
  }
`;

export const StyledHeaderMenu = styled.ul`
  display: flex;
  align-items: center;
  column-gap: 7rem;
`;

export const HeaderMenuItem = styled(NavLink)`
  position: relative;
  ${typography.menu};
  color: ${({ theme }) => theme.colors.primary};
  transition: 0.2s ease-in-out;

  &::before {
    position: absolute;
    content: '';
    bottom: -0.5rem;
    left: 0;
    width: 0;
    height: 2px;
    background-color: ${({ theme }) => theme.colors.primary};
    transition: 0.2s ease-in-out;
  }

  &:hover,
  &:focus-visible {
    &::before {
      width: 100%;
    }
  }
`;
