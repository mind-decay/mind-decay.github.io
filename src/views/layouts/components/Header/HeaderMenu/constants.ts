import { MenuOption } from './types';

import { menuRoutes } from './routes';

export const mainMenuOptions: MenuOption[] = [
  {
    key: menuRoutes.Home,
    value: 'Интро',
  },
  {
    key: menuRoutes.Skills,
    value: 'Навыки',
  },
  {
    key: menuRoutes.Experience,
    value: 'Опыт',
  },
  {
    key: menuRoutes.Contact,
    value: 'Связаться',
  },
];
