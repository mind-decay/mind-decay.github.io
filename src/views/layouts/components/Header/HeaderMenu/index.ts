export * from './HeaderMenu';
export * from './HeaderMenu.styled';
export * from './types';
export * from './routes';
