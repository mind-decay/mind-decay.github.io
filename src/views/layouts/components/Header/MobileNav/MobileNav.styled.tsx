import styled from 'styled-components';

export const MobileNavBackdrop = styled.div<{ $isVisible?: boolean }>`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1000;
  display: ${({ $isVisible }) => ($isVisible ? 'block' : 'none')};
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.colors.backdrop};
`;

export const MobileNavWrapper = styled.aside<{ $isVisible?: boolean }>`
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  z-index: 1001;
  width: 240px;
  padding: 4rem 2rem;
  transition: 0.3s ease-in-out;
  opacity: ${({ $isVisible }) => ($isVisible ? '1' : '0')};
  transform: translateX(${({ $isVisible }) => ($isVisible ? '0' : '100%')});
  background-color: ${({ theme }) => theme.colors.subBG};
`;

export const StyledMobileNav = styled.nav`
  display: flex;
`;

export const MobileNavList = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  row-gap: 1.5rem;
`;
