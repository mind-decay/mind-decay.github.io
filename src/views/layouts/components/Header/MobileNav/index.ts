export * from './MobileNav';
export * from './MobileNav.styled';
