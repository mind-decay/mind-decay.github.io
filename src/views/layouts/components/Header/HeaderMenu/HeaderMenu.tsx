import { ThemeSwitcher } from 'views/components/ThemeSwitcher';
import { HeaderMenuItem, HeaderMenuWrapper, StyledHeaderMenu } from './HeaderMenu.styled';
import { BurgerMenu } from './BurgerMenu';

import { useMedia } from 'hooks';

import { MenuOption } from './types';

import { menuRoutes } from './routes';

import { screenBreakpoints } from 'styles';

export const mainMenuOptions: MenuOption[] = [
  {
    key: menuRoutes.Home,
    value: 'Интро',
  },
  {
    key: menuRoutes.Skills,
    value: 'Навыки',
  },
  {
    key: menuRoutes.Experience,
    value: 'Опыт',
  },
  {
    key: menuRoutes.Contact,
    value: 'Связаться',
  },
];

type HeaderMenuProps = {
  menuOptions: MenuOption[];
  isVisible: boolean;
  onClick: () => void;
};

export const HeaderMenu = ({ menuOptions, isVisible, onClick }: HeaderMenuProps) => {
  const isMobile = useMedia(`(max-width: ${screenBreakpoints.tablet}px)`);

  return (
    <HeaderMenuWrapper>
      {isMobile ? (
        <BurgerMenu isVisible={isVisible} onClick={onClick} />
      ) : (
        <StyledHeaderMenu>
          {menuOptions.map(({ key, value }) => {
            return (
              <li key={key}>
                <HeaderMenuItem to={key}>{value}</HeaderMenuItem>
              </li>
            );
          })}
        </StyledHeaderMenu>
      )}
      <ThemeSwitcher />
    </HeaderMenuWrapper>
  );
};
