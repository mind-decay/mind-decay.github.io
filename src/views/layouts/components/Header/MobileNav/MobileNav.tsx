import { MobileNavBackdrop, MobileNavList, MobileNavWrapper, StyledMobileNav } from './MobileNav.styled';
import { HeaderMenuItem, MenuOption } from '../HeaderMenu';

import { useMedia } from 'hooks';

import { screenBreakpoints } from 'styles';

type MobileNavProps = {
  menuOptions: MenuOption[];
  isVisible: boolean;
  onClick: () => void;
};

export const MobileNav = ({ menuOptions, isVisible, onClick }: MobileNavProps) => {
  const isMobile = useMedia(`(max-width: ${screenBreakpoints.tablet}px)`);

  if (!isMobile) return null;

  return (
    <>
      <MobileNavWrapper $isVisible={isVisible}>
        <StyledMobileNav>
          <MobileNavList>
            {menuOptions.map(({ key, value }) => {
              return (
                <li key={key} onClick={onClick}>
                  <HeaderMenuItem to={key}>{value}</HeaderMenuItem>
                </li>
              );
            })}
          </MobileNavList>
        </StyledMobileNav>
      </MobileNavWrapper>
      <MobileNavBackdrop $isVisible={isVisible} onClick={onClick} />
    </>
  );
};
