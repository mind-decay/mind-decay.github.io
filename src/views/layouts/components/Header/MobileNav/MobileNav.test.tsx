import { vi } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MobileNav } from './MobileNav';

import { mainMenuOptions } from '../HeaderMenu';

test('renders mobile nav', () => {
  const handler = vi.fn();
  const isVisible = true;

  const { container } = render(<MobileNav menuOptions={mainMenuOptions} isVisible={isVisible} onClick={handler} />);

  expect(container).toMatchSnapshot();
});
