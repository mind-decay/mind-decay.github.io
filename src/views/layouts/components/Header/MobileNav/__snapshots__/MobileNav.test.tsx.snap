// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders mobile nav 1`] = `<div />`;
